File: sdks/java/lib/src/main/java/models/shared/TranscriptSkillCard.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * TranscriptSkillCard
 * A Box Skill metadata card that adds a transcript to a file.
**/
public class TranscriptSkillCard {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public TranscriptSkillCard withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("duration")
    public Long duration;
    public TranscriptSkillCard withDuration(Long duration) {
        this.duration = duration;
        return this;
    }
    @JsonProperty("entries")
    public TranscriptSkillCardEntries[] entries;
    public TranscriptSkillCard withEntries(TranscriptSkillCardEntries[] entries) {
        this.entries = entries;
        return this;
    }
    @JsonProperty("invocation")
    public TranscriptSkillCardInvocation invocation;
    public TranscriptSkillCard withInvocation(TranscriptSkillCardInvocation invocation) {
        this.invocation = invocation;
        return this;
    }
    @JsonProperty("skill")
    public TranscriptSkillCardSkill skill;
    public TranscriptSkillCard withSkill(TranscriptSkillCardSkill skill) {
        this.skill = skill;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("skill_card_title")
    public TranscriptSkillCardSkillCardTitle skillCardTitle;
    public TranscriptSkillCard withSkillCardTitle(TranscriptSkillCardSkillCardTitle skillCardTitle) {
        this.skillCardTitle = skillCardTitle;
        return this;
    }
    @JsonProperty("skill_card_type")
    public TranscriptSkillCardSkillCardTypeEnum skillCardType;
    public TranscriptSkillCard withSkillCardType(TranscriptSkillCardSkillCardTypeEnum skillCardType) {
        this.skillCardType = skillCardType;
        return this;
    }
    @JsonProperty("type")
    public TranscriptSkillCardTypeEnum type;
    public TranscriptSkillCard withType(TranscriptSkillCardTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/transcriptskillcard.go
package shared

import (
	"time"
)

// TranscriptSkillCardEntriesAppears
// The timestamp for an entry.
type TranscriptSkillCardEntriesAppears struct {
	Start *int64 `json:"start,omitempty"`
}

// TranscriptSkillCardEntries
// An entry in the `entries` attribute of a metadata card
type TranscriptSkillCardEntries struct {
	Appears []TranscriptSkillCardEntriesAppears `json:"appears,omitempty"`
	Text    *string                             `json:"text,omitempty"`
}

type TranscriptSkillCardInvocationTypeEnum string

const (
	TranscriptSkillCardInvocationTypeEnumSkillInvocation TranscriptSkillCardInvocationTypeEnum = "skill_invocation"
)

// TranscriptSkillCardInvocation
// The invocation of this service, used to track
// which instance of a service applied the metadata.
type TranscriptSkillCardInvocation struct {
	ID   string                                `json:"id"`
	Type TranscriptSkillCardInvocationTypeEnum `json:"type"`
}

type TranscriptSkillCardSkillTypeEnum string

const (
	TranscriptSkillCardSkillTypeEnumService TranscriptSkillCardSkillTypeEnum = "service"
)

// TranscriptSkillCardSkill
// The service that applied this metadata.
type TranscriptSkillCardSkill struct {
	ID   string                           `json:"id"`
	Type TranscriptSkillCardSkillTypeEnum `json:"type"`
}

// TranscriptSkillCardSkillCardTitle
// The title of the card.
type TranscriptSkillCardSkillCardTitle struct {
	Code    *string `json:"code,omitempty"`
	Message string  `json:"message"`
}

type TranscriptSkillCardSkillCardTypeEnum string

const (
	TranscriptSkillCardSkillCardTypeEnumTranscript TranscriptSkillCardSkillCardTypeEnum = "transcript"
)

type TranscriptSkillCardTypeEnum string

const (
	TranscriptSkillCardTypeEnumSkillCard TranscriptSkillCardTypeEnum = "skill_card"
)

// TranscriptSkillCard
// A Box Skill metadata card that adds a transcript to a file.
type TranscriptSkillCard struct {
	CreatedAt      *time.Time                           `json:"created_at,omitempty"`
	Duration       *int64                               `json:"duration,omitempty"`
	Entries        []TranscriptSkillCardEntries         `json:"entries"`
	Invocation     TranscriptSkillCardInvocation        `json:"invocation"`
	Skill          TranscriptSkillCardSkill             `json:"skill"`
	SkillCardTitle *TranscriptSkillCardSkillCardTitle   `json:"skill_card_title,omitempty"`
	SkillCardType  TranscriptSkillCardSkillCardTypeEnum `json:"skill_card_type"`
	Type           TranscriptSkillCardTypeEnum          `json:"type"`
}
